### DISCUSSION TITLE
alternate caption scraper

### DISCUSSION THREAD

**Discussion by tomByrer** [LuanRT/YouTube.js#513] [Ideas]
I saw that you added the ability to [download transcripts via GUI](https://github.com/LuanRT/YouTube.js/pull/500/commits/93d4e1d05d25ccef9cefafc44a659a805b4c0409).  That can work, but incase the transcripts GUI doesn't work, there will likely still be the XML file hidden somewhere that you can grab.  Might be more efficient:
(My special fork; I wanted to keep some HTML markup.  WebVTT output would be easy to add.

https://github.com/tomByrer/youtube-captions-scraper/blob/master/src/index.js

**Comment by absidue** [comment]
YouTube.js already extracts the captions information, it has supported captions for much longer than transcripts.

They also have two different use-cases, which is why YouTube has them separately and YouTube.js supports both. Showing captions on a video where you might want styling or a transcript where you want pure text.

Also there is a reason YouTube calls it a base URL, because you can add a bunch more query parameters to it, YouTube mainly uses it to add tracking parameters but there are two useful ones:
- `fmt`: can be set to `vtt`, `ttml`, `srv1`, `srv2`, `srv3` or `json3`
- `tlang`: can be set to any of the languages in the translation languages array, to make YouTube spit out a machine translation of that caption track, into your desired language.

So if you want a WebVTT file just add `&fmt=vtt` to the captions base URL, no need to do the conversion yourself.
